### PR TITLE
ALTAPPS-1313: Shared, Android skip suggestion

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[ALTAPPS-1313](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1313)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Implement `StepQuizFeedbackState` that represents feedback ViewState;
- Implement `StepQuizFeedbackStateMapper` that map `StepQuizState` into `StepQuizFeedbackStateMapper`;
- Add `wrongSubmissionsCount` into `StepQuizState.AttemptLoaded` which is incremented by `StepQuizFeature.Message.CreateSubmissionSuccess`;
- Add `StepQuizFeature.Action.ViewAction.ShowComments` & `StepQuizFeature.Action.ViewAction.SkipStep`;
- Move Comment toolbar action to `StepMenuSecondaryAction` & remove `StepFeature.ViewState.isCommentsToolbarItemAvailable`.

### CODE REVIEW

File: shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt [hunk 1]
```diff
@@ -619,4 +640,25 @@ internal class StepQuizReducer(
...
+        } else {
+            state to emptySet()
+        }
+    }
+
+    private fun handleReadCommentsClicked(state: State): StepQuizReducerResult =
+        state to setOf(Action.ViewAction.ShowComments)
+
+    private fun handleSkipClicked(state: State): StepQuizReducerResult =
+        state to setOf(Action.ViewAction.SkipStep)
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt:663]: We also need to log analytic events when there is a click on the "see hint," "comments," and "skip" buttons from the feedback. It should be possible to distinguish between these clicks (toolbar and feedback) at the analytics level.

**Review by ivan-magda — APPROVED** [review]

File: shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt (hunk 1 above)
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt]: @ivan-magda 
I fixed it.
Add separate messages for hint, comment & skip request, to not log click analytic event in the origine feature. And log required messages in the StepQuizReducer.

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt]: Thanks 👍

File: shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_hints/presentation/StepQuizHintsFeature.kt [hunk 2]
```diff
@@ -175,6 +175,13 @@ object StepQuizHintsFeature {
         data class ReportHintNoticeHiddenEventMessage(val isReported: Boolean) : Message
     }
 
+    internal sealed class InternalMessage : Message {
+        /**
+         * Initiate loading next hint without analytic event
+         */
+        data object InitiateHintLoading : Message
+    }
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_hints/presentation/StepQuizHintsFeature.kt:183]:
```suggestion
    internal sealed interface InternalMessage : Message {
        /**
         * Initiate loading next hint without analytic event
         */
        data object InitiateHintLoading : InternalMessage
    }
```

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuiz/Views/StepQuizView.swift [hunk 3]
```diff
@@ -311,6 +311,12 @@ private extension StepQuizView {
             assertionFailure(
                 "StepQuizView :: did receive unexpected StepQuizCodeBlanksViewAction: \(stepQuizCodeBlanksViewAction)"
             )
+        case .scrollToHints:
+            #warning("TODO: ALTAPPS-1314")
+        case .showComments:
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuiz/Views/StepQuizView.swift:316]:
```suggestion
        case .requestShowComments:
```

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuiz/Views/StepQuizView.swift [hunk 4]
```diff
@@ -311,6 +311,12 @@ private extension StepQuizView {
...
+            #warning("TODO: ALTAPPS-1314")
+        case .skipStep:
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuiz/Views/StepQuizView.swift:318]:
```suggestion
        case .requestSkipStep:
```

**Pull request merged**